freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/downscale/action/FreeIpaDownscaleActions.java

```diff
@@ -349,6 +353,31 @@ protected void doExecute(StackContext context, StackEvent payload, Map<Object, O
         };
     }
 
+    @Bean(name = "DOWNSCALE_UPDATE_ENVIRONMENT_STACK_CONFIG_STATE")
+    public Action<?, ?> updateEnvironmentStackConfigAction() {
+        return new AbstractDownscaleAction<>(StackEvent.class) {
+            @Inject
+            private EnvironmentEndpoint environmentEndpoint;
+
+            @Override
+            protected void doExecute(StackContext context, StackEvent payload, Map<Object, Object> variables) {
+                Stack stack = context.getStack();
+                stackUpdater.updateStackStatus(stack.getId(), getInProgressStatus(variables), "Updating environment stack config");
+                try {
+                    if (!isRepair(variables)) {
+                        ThreadBasedUserCrnProvider.doAsInternalActor(() -> {
+                            environmentEndpoint.updateConfigsInEnvironmentByCrn(stack.getEnvironmentCrn());
+                        });
+                    }
+                    sendEvent(context, DOWNSCALE_UPDATE_ENVIRONMENT_STACK_CONFIG_FINISHED_EVENT.selector(), new StackEvent(stack.getId()));
+                } catch (Exception e) {
```

Comment:
shouldn't we log the exception here? and if it's a webapplicationexception we might need to use `WebApplicationExceptionMessageExtractor` to get the real message.
same applies to upscale